[sflow + dropmon] added patches for the sFlow + dropmon feature.

vadymhlushko-mlnx opened the pull request:
#### Why I did it
`sFlow + dropmon` feature require the patches provided by SDK team [Mellanox/Switch-SDK-drivers/tree/sx_sdk_4_5_1200_4.5.1208/kernel_backports/5.10.0](https://github.com/Mellanox/Switch-SDK-drivers/tree/sx_sdk_4_5_1200_4.5.1208/kernel_backports/5.10.0)

#### How I did it
Added patches.

saiarcot895 commented:
A few comments:

1. Split the patch you have into two (or more) different patches: one that is backported from upstream (with the commit ID included in the patch), and one that includes any custom changes.
2. In each patch file, describe what changes are being done, with a useful subject line.
3. Is there a reason that the patch has to be conditionally included here, and why it can't be included always?

vadymhlushko-mlnx commented:
> A few comments:
> 
> 1. Split the patch you have into two (or more) different patches: one that is backported from upstream (with the commit ID included in the patch), and one that includes any custom changes.
> 2. In each patch file, describe what changes are being done, with a useful subject line.
> 3. Is there a reason that the patch has to be conditionally included here, and why it can't be included always?

Fixed, could you please review

liat-grozovik commented:
@saiarcot895 kindly reminder to review the changes following comments provided.

saiarcot895 commented on patch/0003-psample-added-define-PSAMPLE_MD_EXTENDED_ATTR.patch:
```diff
@@ -0,0 +1,26 @@
+ 	struct rcu_head rcu;
+ };
+ 
++#define PSAMPLE_MD_EXTENDED_ATTR   1
```
Where is this used?

vadymhlushko-mlnx replied:
Seems like it was never used - deleted patch

vadymhlushko-mlnx commented:
> Please add the upstream commit ID in the description for `0001-psample-Encapsulate-packet-metadata-in-a-struct.patch` and `0002-psample-Add-additional-metadata-attributes.patch`.

Fixed

saiarcot895 commented:
/azp run
